lib/collection/request-auth-base.js

```diff
@@ -0,0 +1,126 @@
+var _ = require('../util').lodash,
+    PropertyBase = require('./property-base').PropertyBase,
+
+    RequestAuthBase;
+
+_.inherit((
+
+    /**
+     * A Postman Request Auth State, which is in charge of holding the intermediate parameters for
+     * the supported auth mechanisms.
+     *
+     * @param {Object} state
+     * @constructor
+     * @extends {PropertyBase}
+     */
+    RequestAuthBase = function PostmanRequestAuthBase (name, state) {
+        // this constructor is intended to inherit and as such the super constructor is required to be executed
+        RequestAuthBase.super_.call(this, arguments);
+
+        _.assign(this, /** @lends RequestAuthBase.prototype */ {
+
+            /**
+             * Name of the auth.
+             *
+             * @type {String}
+             */
+            name: name,
+
+            /**
+             * Holds intermediate state for all authorizations.
+             *
+             * @type {Object}
+             */
+            state: state || {}
+        });
+    }), PropertyBase);
+
+_.extend(RequestAuthBase.prototype, /** @lends RequestAuthState.prototype */ {
+
+    /**
+     * Sets a value in the state.
+     *
+     * @param name
+     * @param value
+     */
+    set: function (name, value) {
```

Comment:
The problem with this is, if someone wants to update the auth parameters itself, they have to do:
```
request.auth.basic.username = 'new-username' // this will be serialized with the collection
```
whereas 
```
request.auth.basic.set('username', 'new-username')  // this will be set in state, so not serialized.
```

Seems rather counter-intuitive in terms of usage.